main.go

```diff
@@ -415,3 +431,23 @@ func installTFProvidedModule(dir string, custBinPath *string) {
 	fmt.Println("No version found to match constraint. Follow the README.md instructions for setup. https://github.com/warrensbox/terraform-switcher/blob/master/README.md")
 	os.Exit(1)
 }
+
+// Install using version constraint from terragrunt file
+func installTGHclFile(tgFile *string, custBinPath *string) {
+	fmt.Printf("Terragrunt file found: %s\n", *tgFile)
+	content := retrieveFileContents(*tgFile)
+	var constraint = ""
+
+	for _, line := range strings.Split(content, "\n") {
+		regex, _ := regexp.Compile(`^terraform_version_constraint\s+=\s+"(?P<version>.*)".*`)
```

Comment:
I think you should move regexp.Compile()  before loop

Reply:
fixed